### DISCUSSION TITLE
RPSL - Interpretation of continuation lines vs multiple instances of the same attribute

### DISCUSSION THREAD

**Discussion by dhutchin-ctl** [irrdnet/irrd#633] [Q&A] [answered]
Consider the following snippit of RPSL submitted to IRRd as part of an RPSL object:

```
remarks:       word1, word2,word3
                    word4, word5,word6
+
+                  word7, word8,word9
remarks:       wordA, wordB,wordC
```

When queried with graphql, the result is:

```
"remarks": [
          "word1, word2,word3,word4, word5,word6,word7, word8,word9",
          "wordA, wordB,wordC"
        ],
```

I understand that '+',  and whitespace in the first column of a line indicates a "continuation" line. but why not treat those the same as a second occurrence of the same attribute.  I don't have anyway in graphql to discover formatting of the original RPSL except through objectText.  I would have thought the above RPSL would have resulted in an array of 5 string elements, the 3rd of which is empty.

I'm not suggesting a change here.  Just curious about why the IRRd implementation was chosen.

**Comment by mxsasha** [comment, marked as answer]
My thoughts here were that RPSL is always a set of key-value pairs, though unlike many structures, the keys are not unique. Therefore, I see this as two key-value pairs. The line split with continuation characters is merely syntactic sugar.

I didn't design much specifically for remarks, as IRRd never concerns itself with their contents. And any parsing anyone else wants to do has to be quite flexible and resilient anyways.
